services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -845,9 +845,10 @@ public static Response redirectAfterSuccessfulFlow(KeycloakSession session, Real
         AuthenticatedClientSessionModel clientSession = clientSessionCtx.getClientSession();
 
         // Update userSession note with authTime. But just if flag SSO_AUTH is not set
-        boolean isSSOAuthentication = "true".equals(session.getAttribute(SSO_AUTH));
+        boolean isSSOAuthentication = "true".equals(authSession.getAuthNote(SSO_AUTH));
         if (isSSOAuthentication) {
             clientSession.setNote(SSO_AUTH, "true");
+            authSession.removeAuthNote(SSO_AUTH);
```

Comment:
I don't think we should remove this note here?

Reply:
If this note is left set then it will interfere with the next login. It is never read elsewhere.